File: app/src/main/java/com/example/borysgurtovyi/codinginterviewprep/HowManyChars.java
package com.example.borysgurtovyi.codinginterviewprep;

import java.util.HashMap;

/**
 * Created by borysgurtovyi on 2/1/18.
 */

public class HowManyChars {

    // str = "Hello, Ultimate Bootcamp!"
    // ['h','e','l','l','o',',',' '....]

    public void howManyChars(String str) {
        HashMap<Character, Integer> myHashMap = new HashMap<>();

        for(char ch : str.toCharArray() ){
            if(!myHashMap.containsKey(ch)) {
                myHashMap.put(ch, 1);
            } else {
                int oldNumberOfCharacter = myHashMap.get(ch);
                int newNumber = oldNumberOfCharacter + 1;
                myHashMap.put(ch, newNumber);
            }
        }

        for(char ch : myHashMap.keySet()) {
            int value = myHashMap.get(ch);
            System.out.println(ch +" => " + value);
        }
    }
}

File: app/src/main/java/com/example/borysgurtovyi/codinginterviewprep/HowManyChars.kt
package com.example.borysgurtovyi.codinginterviewprep

/**
 * Created by lisa on 06.02.2018.
 */

open class HowManyCharsK {
    fun howManyChars(string: String){
        val myMap = HashMap<Char, Int>()

        for(ch in string.toCharArray()){
            myMap[ch] = (myMap[ch] ?: 0) + 1
        }
        for (ch in myMap.keys) {
            val number = myMap[ch]
            println(ch + " ==> " + number)
        }
    }
}